functions/tips-gcp-apis/main.py

```diff
@@ -0,0 +1,49 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# [START functions_tips_gcp_apis]
+import os
+
+from google.cloud import pubsub_v1
+
+
+# Create a global Pub/Sub client to avoid unneeded network activity
+pubsub = pubsub_v1.PublisherClient()
+
+
+def gcp_api_call(request):
+    """
+    HTTP Cloud Function that uses a cached client library instance to
+    reduce the number of connections required per function invocation.
+    Args:
+        request (flask.Request): The request object.
+    Returns:
+        The response text, or any set of values that can be turned into a
+        Response object using `make_response`
+        <http://flask.pocoo.org/docs/1.0/api/#flask.Flask.make_response>.
+    """
+
+    project = os.getenv('GCP_PROJECT')
```

Comment:
Should we add a note this is only for gen 1? Maybe add a link too (https://cloud.google.com/functions/docs/configuring/env-var#python_37_and_go_111)?

Reply:
Done.